project/assets/scripts/controller/evaluations/evaluations.js

```diff
@@ -1,32 +1,38 @@
-const EvaluationsEvents = function () {
-    this.init = function (redirectToFunction) {
-        setNavEvent('new-evaluation', addEvent.bind(null, 'newEvaluation', redirectToFunction));
-        setNavEvent('logout', addEvent.bind(null, 'login', redirectToFunction));
+(function () {
+    const EvaluationsEvents = function () {
+        this.init = function (redirectToFunction) {
+            setNavEvent('new-evaluation', addEvent.bind(null, 'newEvaluation', redirectToFunction));
+            setNavEvent('logout', addEvent.bind(null, 'login', redirectToFunction));
 
-        const allButtonsObject = document.getElementsByTagName('button');
-        const allButtonsArray = Array.prototype.slice.call(allButtonsObject);
-        allButtonsArray.forEach((element) => {
-            const id = element.getAttribute('id');
-            const detailsBtn = document.getElementById(id);
-            detailsBtn.addEventListener('click', redirectToDetails.bind(null, redirectToFunction, detailsBtn.dataset.name));
-        });
+            const allButtonsObject = document.getElementsByTagName('button');
+            const allButtonsArray = Array.prototype.slice.call(allButtonsObject);
+            allButtonsArray.forEach((element) => {
+                const id = element.getAttribute('id');
+                const detailsBtn = document.getElementById(id);
+                detailsBtn.addEventListener('click', redirectToDetails.bind(null, redirectToFunction, detailsBtn.dataset.name));
+            });
+        };
+        this.destroy = function (redirectToFunction) {
+            removeEvents('new-evaluation', setNavEvent('new-evaluation', addEvent.bind(null, 'newEvaluation', redirectToFunction)));
+            removeEvents('logout', addEvent.bind(null, 'login', redirectToFunction));
+
+            const allButtonsObject = document.getElementsByTagName('button');
+            const allButtonsArray = Array.prototype.slice.call(allButtonsObject);
+            allButtonsArray.forEach((element) => {
```

Comment:
Just as an FYI, you should be able to `Array.prototype.forEach.call(allButtonsObject, ...)` instead. Nicely done, nonetheless!